### PR TITLE
feat: add ability to override tsconfig

### LINKED ISSUE #162: [PR WELCOME] Set language level for library sources (es5, es6, es7, dom, webworker, ...)
## Type of Issue

- [ ] Bug Report
- [x] Feature Request

## Description

I have made an application using a custom component library that I have implemented, and now i wanted to use ng-packagr to externalize my custom component library.
The application was created by angular-cli, and i have followed the tutorial of Nikolas LeBlanc and the structure shown in the ng-packaged project.

For some reasons i have used in development es7 features like includes on arrays and padStart on strings. 
With the angular-cli builds including polyfills of core-js everything was working fine but the ng-packagr build gave me errors like this 
```
Property 'includes' does not exist on type 'any[]'.
```

I have tried to include them as externals and import them in the components that required the functions but still have theses errors.

So, how can i add polyfills functions on the build ?

#### Version Information

```
ng-packagr: v1.3.0
node: v7.10.0
@angular: v4.1.3
rxjs: v5.4.2
zone.js: v0.8.14
core-js: v2.4.1
```

### LINKED ISSUE #256: [DRAFT] Add ability to override tsconfig
## Type of Issue

```
[ ] Bug Report
[X] Feature Request
```

## Description

I know that the idea of ng-packagr is to remove as much config as possible from the user but there are cases where being able to manage the tsconfig used by ng-packagr would be very useful, for example adding paths or alternate/additional typeroots. There could be an additional and optional flag to point to the desired tsconfig.

It can cause some difficult to troubleshoot issues when you need to use one config to test your library and another unseen one is used to build.

```
ng-packagr: v1.5.0
node: v8.6.0
@angular: v5.0.0
rxjs: 5.2.2
zone.js: latest
```

Thanks for the awesome work! I might be able to help with a PR if it is something @dherges thinks it is a valuable usecase

### PR DESCRIPTION
**Pull request by DavidParks8**
## I'm submitting a...

```
[ ] Bug Fix
[x] Feature
[ ] Other (Refactoring, Added tests, Documentation, ...)
```

## Checklist

- [x] Commit Messages follow the [Conventional Commits](https://conventionalcommits.org/) pattern
  - A feature commit message is prefixed "feat:"
  - A bugfix commit message is prefixed "fix:"
- [x] Tests for the changes have been added

## Description

fixes #256.
fixes #162.

To configure with a custom `tsconfig` file, put the path relative to your source path in the `tsconfig` field:

```json
{
  "$schema": "./node_modules/ng-packagr/package.schema.json",
  "ngPackage": {
    "tsconfig": "my-custom-tsconfig.json",
    "lib": {
      "entryFile": "public_api.ts"
    }
  }
}
```

The following fields can not be overridden:
- flatModuleId
- flatModuleOutFile
- basePath
- baseUrl
- outDir
- genDir
- target
- module
- moduleResolution
- declaration
- sourceMap
- inlineSources
- allowSyntheticDefaultImports
- skipLibCheck
- emitDecoratorMetadata
- experimentalDecorators
- annotateForClosureCompiler
- skipTemplateCodegen
- strictMetadataEmit

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by deebloo** [comment]
@DavidParks8 what would happen if someone tried to overwrite one of those properties? Should there be some sort of warning (not an error) saying what the user attempted to use and then the value that will actually be used?

**Comment by DavidParks8** [comment]
Essentially, nothing would happen if someone tried to override one of those properties. They are overwritten by the code [here](https://github.com/DavidParks8/ng-packagr/blob/c91e99f6b9892d011ee97072bbd5d3970a02c1dd/src/lib/steps/ngc.ts#L24).

**Comment by DavidParks8** [comment]
I think a warning message would be useful, but I'd rather get this feature out and add the messaging as an additional feature (so those of us who want tsconfig override can use it sooner). In the meantime, the disallowed fields are clearly documented in the readme.

**Comment by DavidParks8** [comment]
@dherges this is ready for review. My team would like it so we can target es2017. Thanks for all the work you’ve put into my-packagr.

File: .circleci/config.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       # Run the ci build
       - run: yarn install --frozen-lockfile --non-interactive
       - run: yarn test
-      # Store artefacts on circle
+      # Store artifacts on circle
```
Comment (dherges) [inline .circleci/config.yml]: In lovel britisih english it's "artefacts".

**Comment by dherges** [comment]
The use case is es2017 language level support.

Internal / implementation detail: tsconfig `compilerOptions.lib = [ /*..*/ es2017 /*..*/ ]) 
Usage: `ngPackage.lib.languageLevel = "es2017"`

Do not send a pull request. Wait for a 2.0.0 release.

**Comment by DavidParks8** [comment]
It isn’t just language level we desire, it’s also much stricter compiler options, as well as custom paths. Language level was just one use case.

**Comment by DavidParks8** [comment]
Are you not wanting this feature flat out? Or are you just saying to wait until 2.0, then PR again?

**Comment by dherges** [comment]
I'm in a hurry. Can only write short sentences: already had discussion about language level this week and we agreed to do a languageLevel property. Wait until 2.0, there are more changes coming and it's going to conflict all over the place.

**Comment by DavidParks8** [comment]
Will do.

**Comment by DavidParks8** [comment]
I didn’t know about the British spelling. Thanks for the clarification. If desired, I can implement this feature without the spelling change after 2.0.0 rtm. Take your time in responding. Ideally, I’d like the ability to use new ts features as they come out without waiting for an ng-packagr release, angular peer dependency restrictions permitting of course.

**Comment by rmeshksar** [comment]
I would like to disable annotateForClosureCompiler to false as it creates some issues using typescript 2.4.2 and string enums.
This code would satisfy my needs. Will it be merged as part of v2.0.0?

**Comment by ghaiat** [comment]
Hello
is it possible npow to turn off strictMetadatacheck ? otherwise static class are failing

**Pull request closed without merging**